makefiles/multislot.mk

```diff
@@ -51,4 +52,136 @@ else
 mcuboot:
 	$(Q)echo "error: mcuboot not supported on board $(BOARD)!"
 	$(Q)false
+
 endif # SLOT0_SIZE
+endif # MAKECMDGOALS
+
+ifneq (,$(filter multislot combined flash-multislot keys verify bootloader multislot-clean, $(MAKECMDGOALS)))
+
+ifndef SLOT0_SIZE
+$(error Board $(BOARD) does not define multislot parameters!)
+endif
+
+ELFBOOTLOADER := $(ELFFILE)
+ELFSLOT1      := $(ELFFILE:%.elf=%.slot1.elf)
+ELFSLOT2      := $(ELFFILE:%.elf=%.slot2.elf)
+
+BINBOOTLOADER := $(ELFFILE:%.elf=%.bin)
+BINSLOT1      := $(ELFSLOT1:%.elf=%.bin)
+BINSLOT2      := $(ELFSLOT2:%.elf=%.bin)
+
+COMBINED_BIN := $(ELFFILE:%.elf=%-slot1-$(FW_APPID)-$(FW_VERSION).combined.bin)
+IMAGE_SLOT1 := $(ELFFILE:%.elf=%-slot1-$(FW_APPID)-$(FW_VERSION).bin)
+IMAGE_SLOT2 := $(ELFFILE:%.elf=%-slot2-$(FW_APPID)-$(FW_VERSION).bin)
+
+FIRMWARE_METADATA_SIZE ?= 256
+SLOT0_ADDR_DEC = $(shell printf "%d" $(SLOT0_ADDR))
+SLOT0_SIZE_DEC = $(shell printf "%d" $(SLOT0_SIZE))
+SLOT1_SIZE_DEC = $(shell printf "%d" $(SLOT1_SIZE))
+FIRMWARE_METADATA_SIZE_DEC = $(shell printf "%d" $(FIRMWARE_METADATA_SIZE))
+SLOT1_ADDR = $(shell echo 'obase=16; ${SLOT0_ADDR_DEC}+${SLOT0_SIZE_DEC}+${FIRMWARE_METADATA_SIZE_DEC}' | bc)
+SLOT2_ADDR = $(shell echo 'obase=16; ${SLOT0_ADDR_DEC}+${SLOT0_SIZE_DEC}+${SLOT1_SIZE_DEC}+${FIRMWARE_METADATA_SIZE_DEC}' | bc)
+
+FIRMWARE_TOOLS = $(RIOTBASE)/dist/tools/firmware_metadata
```

Comment:
Rename in FIRMWARE_TOOLS_DIR and use that variable in the lines below to avoid duplication